intro-unal-2022-1-tanks/Assets/Scr/TurretMovement.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+public class TurretMovement : MonoBehaviour
+{
+    [SerializeField] private Transform jugador;
+    [SerializeField]  private Transform[] turrets;
+    
+    // Objetos vacíos para el movimiento de la torreta
+    [SerializeField] private Transform point1; // Objeto P1
+    [SerializeField] private Transform point2; // Objeto P2
+    public float speed = 0.2f;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+    
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        // Movimiento de Torreta entre la posicion de dos objetos vacíos point1 (P1) y point2 (P2)
+        transform.position = Vector3.Lerp (point1.position, point2.position, Mathf.PingPong(Time.time*speed, 1.0f));
+ 
+        // Torretas apuntan al jugador
+        Vector3 dir = jugador.position - transform.position;
```

Comment:
Perfecto!
En caso de que se quiera más presición, este cálculo se deberia hacer dentro del `for` y usando el transform de cada torreta en vez del transform del padre. Así tomas en cuenta ese desfase que tiene cada torreta respecto al (0, 0) del padr.